src/app/profile/yourGiving/stopStartRecurringGifts/restartGift/restartGift.component.js

```diff
@@ -166,6 +185,9 @@ class RestartGiftController {
       case 'confirm':
         this.step = 'configure';
         break;
+      case 'add-update-payment-method':
+        this.changeState( {state: 'step-0'} );
```

Comment:
Should this go back to step0PaymentMethodList if `hasPaymentMethods`?  That should mean they came from the list to this state. If they don't have payment methods, going to step-0 should be good. (That's the list of Restart, Change, Redirect, Stop right?)